godtossing/Pos/Ssc/GodTossing/Toss/Base.hs

```diff
@@ -483,12 +484,17 @@ checkCertificatesPayload epoch certs = do
     exceptGuardSnd CertificateDuplicateVssKey
         (not . (`HS.member` existingVssKeys) . vcVssKey)
         (HM.toList (getVssCertificatesMap certs))
-    let newVssKeys =
-            HS.fromList . map vcVssKey . toList . getVssCertificatesMap $
-            certs
-    exceptGuardSnd CertificateDuplicateVssKey
-        (not . (`HS.member` newVssKeys) . vcVssKey)
-        (HM.toList (getVssCertificatesMap certs))
+    -- find pubkeys corresponding to vss keys that are duplicated
+    let isDupGroup (_:_:_) = True
+        isDupGroup _       = False
+    let keysWithDups =
```

Comment:
it seems this check won't work b/c two *different* stakeholder can't have two same certificates because vss certificate includes _public key_ and default `Eq` instance for `VssCertificate` will say vss certificate have different public keys, hence they aren't equal.
so we should rewrite `Eq` for `VssCertificate` or something like that.

Reply:
ok, I am wrong